contribs/gmf/src/services/editfeature.js

```diff
@@ -26,21 +27,36 @@ gmf.EditFeature = function($http, gmfLayersUrl) {
   this.http_ = $http;
 
   /**
-   * @type {string}
+   * @type {angular.$q}
    * @private
    */
-  this.baseUrl_ = gmfLayersUrl;
+  this.q_ = $q;
+
+  /**
+   * Url to the GeoMapFish layers service. Required in applications that use:
+   * - the editfeature tools
+   * - the objectediting tools
+   *
+   * @type {?string}
+   * @private
+   */
+  this.baseUrl_ = $injector.has('gmfLayersUrl') ?
+    $injector.get('gmfLayersUrl') : null;
 
 };
 
 
 /**
+ * Build a query to the MapFish protocol to fetch features from a list
+ * of layer ids inside a specified extent.
+ *
  * @param {Array.<number>} layerIds List of layer ids to get the features from.
  * @param {ol.Extent} extent The extent where to get the features from.
  * @return {angular.$q.Promise} Promise.
  * @export
  */
-gmf.EditFeature.prototype.getFeatures = function(layerIds, extent) {
+gmf.EditFeature.prototype.getFeaturesInExtent = function(layerIds, extent) {
+  goog.asserts.assert(this.baseUrl_, 'GMF layers url must be defined');
```

Comment:
I kinda disagree.  The property to set is `gmfLayerUrl`, which is used in the `gmf.EditFeature`.  It seems simpler to manage where's it's needed instead of one level above, in my opinion.  What do you think?